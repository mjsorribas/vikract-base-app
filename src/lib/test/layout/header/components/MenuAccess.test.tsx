import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { expect, test, vi } from "vitest";

import MenuAccess from "lib/layout/header/components/menuaccess/menuAccess";

// Mock de una dependencia especÃ­fica
vi.mock("lib/layout/header/components/menuaccess/menuAccess", () => ({
  __esModule: true,
  default: () => <div>Mocked MenuAccess</div>,
}));

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
};

test("Menu Access", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={[{ pathname: "/" }]}>
      <MenuAccess />
    </MemoryRouter>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
