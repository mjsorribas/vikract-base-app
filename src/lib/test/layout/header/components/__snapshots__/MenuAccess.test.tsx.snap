// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Menu Access 1`] = `
<div
  className="css-0"
>
  <div
    className="css-0"
  >
    <div
      className="chakra-stack css-111rh5x"
    >
      <button
        className="chakra-button css-4xx2wk"
        type="button"
      >
        Sign Up
      </button>
      <button
        className="chakra-button css-4xx2wk"
        type="button"
      >
        Sign In
      </button>
    </div>
  </div>
</div>
`;
